### ISSUE TITLE
snapcraft 7 ignores icon path in snapcraft.yaml

### ISSUE THREAD

**Issue by mapero** [canonical/snapcraft#4936]
### Bug Description

I have minimal example of a snapcraft.yaml
```
name: icon-snap
base: core22
version: '0.1'
summary: icon snap
icon: icon/icon.png
description: minimal
grade: stable
confinement: strict
parts:
  my-part:
    plugin: nil
```

It seems like the icon is not copied when there is no "apps" section defined in the snapcraft.yaml in snapcraft 7. With snapcraft 6 or legacy the icon is copied into the snap:

With snapcraft 6 / legacy
```
prime/
└── meta
    ├── gui
    │   └── icon.png
    └── snap.yaml

2 directories, 2 files
```

With snapcraft 7
```
prime/
└── meta
    ├── gui
    └── snap.yaml

2 directories, 1 file
```

### To Reproduce

Build the following snap with snapcraft 7:

```
name: icon-snap
base: core22
version: '0.1'
summary: icon snap
icon: icon/icon.png
description: minimal
grade: stable
confinement: strict
parts:
  my-part:
    plugin: nil
```

### Environment

Ubuntu 22.04 LTS

### snapcraft.yaml

```shell
name: icon-snap
base: core22
version: '0.1'
summary: icon snap
icon: icon/icon.png
description: minimal
grade: stable
confinement: strict
parts:
  my-part:
    plugin: nil
```

### Relevant log output

```shell
NA
```

**Comment by PsikoBlock** [comment]
Relevant source: 
https://github.com/canonical/snapcraft/blob/8c2c801e7beda637e185e576969fd8ffaa9f76d7/snapcraft/parts/setup_assets.py#L75-L80
Lines 75 and 76 should be moved below the call to _finalize_icon to match Snapcraft 6's behavior. This is unchanged in Snapcraft 8.

**Comment by mr-cal** [comment]
Hi @mapero and @PsikoBlock,

Thanks for the bug report and root cause.  This indeed looks like a regression.

I have proposed a fix in #4950.  Do you need this fix in snapcraft 7?  Or is it ok to just fix this in snapcraft 8?

**Comment by mapero** [comment]
Hi @mr-cal ,

we would need this also fixed in snapcraft 7 since we will stick with snapcraft 7 for quite some time. 

Thanks.

**Comment by mr-cal** [comment]
Sounds good, we will target both snapcraft 7 and 8.

**Comment by mr-cal** [comment]
Fixed via #4950.  I'll snapcraft 7.5.6 this week or next week.

**Comment by mr-cal** [comment]
Released to `7.x/stable` for everything except riscv64 (see #4980)
